File: Challenges with Solutions/sets-cherelemma-main/challenge.js
function findDuplicates(array) {
  // write your code here
  const mySet = new Set();
  const duplicates = array.filter((item) => {
      if(mySet.has(item)) {
        return true;
      }
      mySet.add(item)
      return false;
    });
  
  /* const duplicates = [];
  for (let i = 0; i < array.length; i = i + 1) {
    mySet.has(array[i]) ? duplicates.push(array[i]): mySet.add(array[i]);
  }
  */
  
  return duplicates;
}

console.log(findDuplicates([1, 2, 3, 1, 5, 6, 7, 8, 5, 2]))
// => [1, 5, 2]

console.log(findDuplicates([3, 501, 17, 23, -43, 67, 5, 888, -402, 235, 77, 99, 311, 1, -43]))
// => [-43]

module.exports = findDuplicates

File: Challenges with Solutions/quick-sort-partition-cherelemma-main/challenge.rb
def partition(array)
  # write your code here
  pivot = array[0]
  pindex = 0
  
  (1...array.length).each do |i|
    if pivot >= array[i]
      j = i
      while j > pindex do
        array[j], array[j-1] = array[j-1], array[j]
        j -= 1
      end
      pindex += 1
    end
  end
 return array
end

p partition([4, 5, 3, 9, 1])
# => [3, 1, 4, 5, 9]
p partition([2, 10, 3, 7, 9, 4, 6, 12, 8])
# => [3, 1, 4, 5, 9]